request.go

```diff
@@ -140,6 +140,91 @@ type Request struct {
 	cancelCtx context.CancelFunc
 }
 
+// This is essentially a more unmarshaled Request.
+// I say more unmarshaled, because StatAttrs is a FileStat, which while better
+// than a binary array, is still supposed to be sftp protocol values instead of
+// something like os or fs package types and values.
+type ParsedRequest struct {
+	Method    string
+	Filepath  string
+	OpenFlags FileOpenFlags
+	StatFlags FileAttrFlags
+	StatAttrs FileStat
+	Target    string // for renames and sym-links
+}
+
+// Marshal is used to turn a ParsedRequest into a Request.
+func (pr *ParsedRequest) Marshal() (r *Request) {
```

Comment:
Agreed.  Though I think that it's moot given the pattern you mentioned for tackling this without a new object at all.

Reply:
I tend to leave some comments even on code I’m suggesting that should change, because there can be value gleaned from them outside of just improving the current code.

Reply:
Much appreciated. :)